vcf.c

```diff
@@ -4699,6 +4700,16 @@ bcf_hdr_t *bcf_hdr_merge(bcf_hdr_t *dst, const bcf_hdr_t *src)
                 if (res < 0) return NULL;
                 need_sync += res;
             }
+            else if ( !strcmp(src->hrec[i]->key,"fileformat") )
+            {
+                int ver_src = bcf_get_version(src,src->hrec[i]->value);
+                int ver_dst = bcf_get_version(dst,dst->hrec[j]->value);
+                if ( ver_src > ver_dst )
+                {
+                    res = bcf_hdr_set_version(dst,src->hrec[i]->value);
+                    need_sync += res;
```

Comment:
`bcf_hdr_set_version` returns 0 on success and <0 on error (eg strdup allocation failure).  So it's inappropriate to do `need_sync += res` as at best it'll do nothing at all, or on error it'll even *remove* the need to syncronise.

We should change that to `need_sync = 1`.